prov/psm3/configure.m4

```diff
@@ -66,16 +66,15 @@ ifelse('
 
 		AC_MSG_CHECKING([for -msse4.2 support])
 		save_CFLAGS=$CFLAGS
-		CFLAGS="$CFLAGS -msse4.2"
+		CFLAGS="$CFLAGS -msse4.2 -O0"
```

Comment:
What if CFLAGS includes `-Osomethingelse`?  How do you know which one will be used?

Reply:
The last -O? argument in the command line is used.

Reply:
Is that documented behavior in all the compilers that we support?

Reply:
I did not see the list of supported compilers, but either way, it looks like it is safer to use the function in the m4 file linked by @shefty and just strip away all the optimization flags to test properly.